azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/DependencyInjectionContainerHolder.kt

```diff
@@ -3,14 +3,34 @@
 
 package com.azure.android.communication.ui.presentation
 
-import androidx.lifecycle.ViewModel
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
 import com.azure.android.communication.ui.di.DependencyInjectionContainer
+import com.azure.android.communication.ui.di.DependencyInjectionContainerImpl
+import java.lang.RuntimeException
 
-internal class DependencyInjectionContainerHolder : ViewModel() {
-    var container: DependencyInjectionContainer? = null
+/**
+ * ViewModel for the CallCompositeActivity
+ *
+ * The instanceID should match the one passed to the Activity via Intent.
+ * it needs to be set in activity onCreate() before accessing container or it
+ * will throw an exception.
+ *
+ * Afterwards you can reference container, which holds the services.
+ */
+internal class DependencyInjectionContainerHolder(application: Application) : AndroidViewModel(application) {
+    // / Instance ID to locate Configuration. -1 is invalid.
+    var instanceId: Int = -1
 
-    override fun onCleared() {
-        super.onCleared()
-        container = null
+    val container: DependencyInjectionContainer by lazy {
+        if (instanceId == -1) {
+            throw RuntimeException(
+                "Will not be able to locate a Configuration for instanceId: -1. " +
+                    "Please ensure that you have set instanceId before retrieving the container."
+            )
+        }
+
+        // / Generate a new instance
```

Comment:
The code is very clean, I do not think we need comment :)

Reply:
removed